src/litserve/server.py

```diff
@@ -429,6 +430,12 @@ def __init__(
                 "Please provide a valid api path like '/predict', '/classify', or '/v1/predict'"
             )
 
+        # TODO: discuss this condition and the warning message with the team
+        if "batch" in lit_api.__class__.__dict__ and "unbatch" in lit_api.__class__.__dict__ and max_batch_size == 1:
```

Comment:
LitAPI always has a default `batch` and `unbatch` method so this condition might be always true. We would also need to check if the user has implemented `batch` and `unbatch`. We can do something like [here](https://github.com/Lightning-AI/LitServe/blob/4d5d1466ad4c43a87c189ac418ef179ae5b508e0/src/litserve/api.py#L165).